WIP: Private key management chapter (#48)
BitcoinDesign/Guide: guide/private-key-management/single-user-schemes.md

This can raise the security since anyone needs access to more than one key to move any funds. But clearly it also increases complexity and the requirements on the user to keep even more keys securely stored and/or backed up.

*{illustration}*

**How it works** - A software wallet-product or coordination software initiates a multi-sig wallet that can have several private keys attached (co-signers), the user then adds private keys from other wallets generated elsewhere to the multisig after which the software wallet can complete the creation process. For any future transaction from the multi-sig wallet the required amount of co-signers need to sign (using Partially Signed Bitcoin Transactions - PSBT from [BIP174]) before any transaction is valid.

**Pros** - Can add redundancy to private keys and increase security

**Cons** - Adds complexity and op-sec burden for multiple private keys

> **When to use** - When storing large amounts, when users are likely to be very knowledgeable

Reviewer: What counts as "very knowledgable"? I think this can be made more specific for example:

```suggesstion
**When to use** - When storing large amounts, when users are confident and able to manually safeguarding their private keys.
```

   retina = "/assets/images/guide/private-key-management/schemes/personal-schemes@2x.jpg"
   alt-text = ""
   caption = "Cloud backup, manual backup , signing-device, threshold-signatures and multi-key."
   width = 2012
   height = 400
%}

- [Automatic cloud backup]({{ '/guide/private-key-management/single-user-schemes/#automatic-cloud-backup' | relative_url }})
- [Manual backup / Recovery phrase]({{ '/guide/private-key-management/single-user-schemes/#recovery-phrase' | relative_url }})
- [External signing device]({{ '/guide/private-key-management/single-user-schemes/#external-signing-device' | relative_url }})
- [Threshold signatures / Key-sharing]({{ '/guide/private-key-management/single-user-schemes/#threshold-signatures--key-sharing' | relative_url }})
- [Multi-key]({{ '/guide/private-key-management/single-user-schemes/#multi-key' | relative_url }})

***

## Automatic cloud backup

By using an automatic cloud backup for the private key the user is NOT required to manually backup anything when setting up a new wallet. This is sometimes also called *seedless*, as the user is never exposed to the traditional *seed*, or recovery-phrase that is used to initiate a wallet. 

> The target audience with this scheme would be new or novice bitcoin users to whom being exposed to backing up a recovery-phrase as part of onboarding can be baffling and backfire. It could also be a starting point for a progressive security model that lets customers upgrade when the start accumulating more funds. 

Reviewer 2: ```suggestion
The target audience with this scheme would be new or novice bitcoin users to whom being exposed to backing up a recovery-phrase as part of onboarding can be baffling and backfire. It could also be a starting point for a progressive security model that lets customers upgrade when they start accumulating more funds. 
```

The automatic cloud backup scheme gives users reasonable security against theft, while providing low risk for self-inflicted losses. 

{% include image.html
   image = "/assets/images/guide/private-key-management/schemes/cloud-backup.jpg"
   retina = "/assets/images/guide/private-key-management/schemes/cloud-backup@2x.jpg"
   alt-text = "Automatic cloud backup illustration"
   caption = "Private keys are encrypted and stored in the cloud."
   width = 800
   height = 400
%}

#### Pros 
- Low onboarding friction
- Low risk of self-inflicted loss due to the automatic nature and redundant backups

#### Cons
- Private key in cloud storage increases risk of malicious third party access
- User needs to trust their OS and cloud storage provider
- Unless the backup includes the recovery phrase users will not be able to export their wallet to another product

### How it works

> The wallet-application will generate a wallet with a keypair, then encrypt and back the key up to a location that the user is unlikely to lose access to, while hard for a malicious third party to gain access to. 

Reviewer 2: Since this is significant paragraph, I propose it's broken down into simple  to understand sentences that explain process. Suggestion below, maybe that's now what you're looking for.

```suggestion
First the application generates a wallet with a keypair. Then, the key is encrypted and backed up to a location convenient and safe for the user, yet hard to for a malicious third-party to gain access to.

```

> With most implementations so far, the location will be the keychain or key-value store on a mobile device that is also backed up by the OS provider such as iOS (iCloud) or Android (Google Drive). 

Reviewer 2: Typo.
```suggestion
With most implementations so far, the location will be the keychain or key-value stored on a mobile device that is also backed up by the OS provider such as iOS (iCloud) or Android (Google Drive). 
```

Author: Not a typo, but I get it's a bit too technical. Will adjust.

This makes the backup accessible by the user on a new device, should they lose the original, but only accessible by someone that can log into the user’s Apple or Google account. 

### Best practice

**When to use** 
- When risk of self-inflicted loss is higher than loss from theft
- For beginners, or products with progressive security
- For lower values

**When not to use** 
- When storing high values
- When risk of loss from theft is higher than self-inflicted loss
- When complete self-sovereignty is important to the target audience

**Variations** 
- Storage location (keychain, iCloud, Google Drive, One Password etc.)
- Additional user authentication (biometrics, pin)

**Products that use this scheme** 
- [Photon SDK]({{ 'https://photonsdk.org'}}) - for use by other wallets, in development
- [Casa]({{ 'https://keys.casa'}})
- [Pine]({{ https://pine.pm }}) - Lightning, beta

> - [Paymobil]({{ 'https://www.paymobil.co'}}) - not a bitcoin wallet

Reviewer 2: > paymobil

Curious, how is it related to the use-case here?

Author: Will remove as it's not a bitcoin product.

> - Bitcoin.com mobile wallet

Reviewer 2: Do we really want to mention the website that's been scamming users for years now?

Author: Deleted

***

## Manual backup / Recovery-phrase

Manual backup of the recovery phrase have been the most common private key management scheme by far, since its proposal with [BIP39]({{ 'https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki'}}) in 2013. If you have used any non-custodial bitcoin product you are likely to have experienced the onboarding requirements of manual backups. 

> After creating a new wallet, you will be asked to manually backup a 12 (or 24 in rare instances) word recovery-phrase to a *safe place*. As the next step it will often ask you to verify that you did save it by having you input the phrase in the correct order. 

Reviewer 2: >  12 (or 24 in rare instances) 

Do we have any data on this? How is it a rare instance? Personally I encountered 24 far more often. Maybe just say 12 or 24.

Author: Changing

   caption = "Recovery-phrases should be stored off-line on metal or paper backups."
   width = 800
   height = 400
%}

#### Pros 
- Manual backups done well can provide very high security
- Good interoperability

#### Cons 
- Requires significant effort from users to achieve safe and redundant backups
- High onboarding friction

### How it works
The wallet-application will generate a 12 or 24 word phrase (seed / recovery phrase) from which all the wallet’s keys can be derived. This means the user can have access to the wallet from any compatible wallet-application with the phrase, even if they lose the device or software.

This can be an effective way to reduce the risk of loss from theft if the backup is off-line in a safe place, but puts more of the op-sec burden on the individual user. The security and risk will only be as good as how they backup the recovery phrase. To get the best security out of recovery phrases they should be in several redundant and secure places off-line. You also need to take resistance to physical theft, fire and other destructive events into account. A fair amount of effort and knowledge is required for this. 

> A full guide to manual backup best practices for users is out of scope for this chapter but the [Do's and Dont's of Bitcoin key management]({{ https://blog.keys.casa/the-dos-and-donts-of-bitcoin-key-management/ }}) is a good starting point.

Reviewer 2: This link doesn't work locally, redirects to this page.

Author: Confirmed working in the preview build linked from GitHub

{% include image.html
   image = "/assets/images/guide/private-key-management/schemes/threshold-signatures.jpg"
   retina = "/assets/images/guide/private-key-management/schemes/threshold-signatures@2x.jpg"
   alt-text = "Threshold signatures illustration"
   caption = "One key is split into several pieces, more than one are required for controlling the wallet."
   width = 800
   height = 400
%}

#### Pros
- Can provide higher resistance to loss from theft
- Transactions look identical and have same cost as single key wallets on chain

#### Cons 
- Requires precise coordination of key-shares when signing
- Few advantages over multi-key setups with Schnorr signatures
- Individual product implementations not interoperable

### How it works

> A single private key is split into n key-shares that are distributed to several parties, devices or locations. When signing a transaction the required number of shares need to be coordinated into one valid signature. This relies on a cryptographic algorithm called [Shamir's Secret Sharing]({{ 'https://en.wikipedia.org/wiki/Shamir's_Secret_Sharing'}}) after it's creator.

Author: Opens to the correct page on preview build linked from GitHub

%}

# Personal schemes

Now let's dive in and look at the various schemes that might be suitable for a personal wallet application with a single user to understand the differences at a high level.

{% include image.html
   image = "/assets/images/guide/private-key-management/schemes/personal-schemes.jpg"
   retina = "/assets/images/guide/private-key-management/schemes/personal-schemes@2x.jpg"
   alt-text = ""
   caption = "Cloud backup, manual backup , signing-device, threshold-signatures and multi-key."
   width = 2012
   height = 400
%}

- [Automatic cloud backup]({{ '/guide/private-key-management/single-user-schemes/#automatic-cloud-backup' | relative_url }})
- [Manual backup / Recovery phrase]({{ '/guide/private-key-management/single-user-schemes/#recovery-phrase' | relative_url }})
- [External signing device]({{ '/guide/private-key-management/single-user-schemes/#external-signing-device' | relative_url }})
- [Threshold signatures / Key-sharing]({{ '/guide/private-key-management/single-user-schemes/#threshold-signatures--key-sharing' | relative_url }})
- [Multi-key]({{ '/guide/private-key-management/single-user-schemes/#multi-key' | relative_url }})

***

> ## Automatic cloud backup

Reviewer 3: Do you think it's helpful what is meant by cloud backup? To me it always sounds like some online storage, but in reality it's the keychain (or password manager).

***

## External signing device

Most common are hardware wallets who’s makers often provide their own software to pair with. But there are also third party wallet applications that support external signing, or software that can be run on off-line (air gapped) computers to perform the same function. 

A wallet application that supports external signing can remove the need for the private key to be readable by the wallet software, thereby decreasing the risk of keeping keys on a device that is likely always connected to the internet.

{% include image.html
   image = "/assets/images/guide/private-key-management/schemes/signing-devices.jpg"
   retina = "/assets/images/guide/private-key-management/schemes/signing-devices@2x.jpg"
   alt-text = "External signing device illustration"
   caption = "Hardware wallets are the most common external signing devices."
   width = 800
   height = 400
%}

#### Pros 
- Removes private key from online devices
- Can provide very high security if used correctly

#### Cons
- Purchase of specialized hardware required
- Requires further effort and knowledge to setup correctly
- Still requires a good manual backup for redundancy

## How it works

> The external device can generate and store a keypair offline, the private key has thus never been known outside the device. A softare wallet can connect to the signing device and act as an interface, or handle transactiosn airgapped from the device. 

Reviewer 3: ```suggestion
The external device can generate and store a keypair offline, the private key has thus never been known outside the device. A software wallet can connect to the signing device and act as an interface, or handle transactiosn airgapped from the device. 
```

Reviewer: ```suggestion
The external device can generate and store a keypair offline, the private key has thus never been known outside the device. A software wallet can connect to the signing device and act as an interface, or handle transactions airgapped from the device. 
```

misspelling:
- transactiosn

When a transaction needs to be signed, the software submits a partially signed bitcoin transaction (PSBT, defined in [BIP174]('https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki'){:target="_blank"}) to the device. The user confirms on the device and the PSBT state of the transaction returned to the software wallet application is now fully signed and can be transmitted to the blockchain. This process can also happen fully airgapped by using memory cards instead of cables or wireless connections, to keep the device offline at all times.

### Best practice

**When to use** 
- When storing medium to large amounts
- When risk of loss from theft is higher than loss from negligence 

> - When target audience is likely to be very knowledgeable or motivated to learn good manual backup schemes, 

Reviewer 3: ```suggestion
- When target audience is likely to be very knowledgeable or motivated to learn good manual backup schemes
```

**When not to use** 
- When most users are new to bitcoin and unlikely to implement good backup schemes
- When transactions are likely to be frequent and low value

**Variations** 
- Interfaces (bluetooth, USB, microSD card, camera, NFC)

> - Platform compatability

Reviewer 3: ```suggestion
- Platform compatibility
```

{% include image.html
   image = "/assets/images/guide/private-key-management/schemes/threshold-signatures.jpg"
   retina = "/assets/images/guide/private-key-management/schemes/threshold-signatures@2x.jpg"
   alt-text = "Threshold signatures illustration"
   caption = "One key is split into several pieces, more than one are required for controlling the wallet."
   width = 800
   height = 400
%}

#### Pros
- Can provide higher resistance to loss from theft
- Transactions look identical and have same cost as single key wallets on chain

#### Cons 
- Requires precise coordination of key-shares when signing
- Few advantages over multi-key setups with Schnorr signatures
- Individual product implementations not interoperable

### How it works

> A single private key is split into n key-shares that are distributed to several parties, devices or locations. When signing a transaction the required number of shares need to be coordinated into one valid signature. This relies on a cryptographic algorithm called [Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir's_Secret_Sharing){:target="_blank"} after it's creator.

Reviewer 3: ```suggestion
A single private key is split into n key-shares that are distributed to several parties, devices or locations. When signing a transaction the required number of shares need to be coordinated into one valid signature. This relies on a cryptographic algorithm called [Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir's_Secret_Sharing){:target="_blank"} after its creator.
```

### Best practice

**When to use** 
- When target audience is knowledgeable 

> - When risk of theft is higher than self-inflected loss

Reviewer 3: ```suggestion
- When risk of theft is higher than self-inflicted loss
```
